examples/simple-game-server/main.go

```diff
@@ -36,7 +36,8 @@ import (
 
 // main starts a UDP or TCP server
 func main() {
-	go doSignal()
+	sigCtx, sigCancel := context.WithCancel(context.Background())
```

Comment:
Rather than keeping `doSignal`, could we simplify and remove that function, and instead do:

```
ctx := signals.NewSigKillContext()
```

And make a minor change to the section below (line 125) of:

```
	<-ctx.Done()
	os.Exit(0)
```

and we don' t need to worry about wrapping contexts or anything like that.

WDYT?

Reply:
@markmandel 
I agree with that.
I'll fix it.

But var `ctx` is already used at the below lines to manage health context.
```
ctx, cancel := context.WithCancel(context.Background())
go doHealth(s, ctx)
```

So should we use `sigCtx` to manage signal context like the one below?
```
func main() {
	sigCtx := signals.NewSigKillContext()

        ~~~//other tasks
        
        <-sigCtx.Done()
	os.Exit(0)
}
//doSignal is erased.
```

Or should we rename `ctx` which is used to `go doHealth(s, ctx)` like `healthCtx` then keep using `ctx` on main like that?
```
func main() {
	ctx := signals.NewSigKillContext()

        ~~~
        healthCtx, cancel := context.WithCancel(context.Background())
	go doHealth(s, healthCtx)
        ~~~
        
        <-ctx.Done()
	os.Exit(0)
}
//doSignal is erased.
```

Reply:
> Or should we rename ctx which is used to go doHealth(s, ctx) like healthCtx then keep using ctx on main like that?

Either way, I don't mind. I'll leave the decision up to you 🤗